src/main/io/gps.c

```diff
@@ -686,77 +1100,144 @@ void gpsInitUblox(void)
                 break;
             }
 
+            DEBUG_SET(DEBUG_GPS_UNIT_CONNECTION, 5, gpsData.state_position);
+
             if (gpsData.ackState == UBLOX_ACK_IDLE) {
                 switch (gpsData.state_position) {
+                    // if a UBX command is sent, ack is supposed to give position++ once the reply happens
+                    case UBLOX_DETECT_UNIT:
+                        if (gpsData.unitVersion == UBX_VERSION_UNDEF) {
+                            ubloxSendClassMessage(CLASS_MON, MSG_MON_VER, 0);
+                            break;
+                        }
+                        gpsData.state_position++;
+                        break;
                     case UBLOX_INITIALIZE:
-                        gpsData.ubloxUsePVT = true;
-                        gpsData.ubloxUseSAT = true;
+                        ubloxSetNavRate(1, 1, 1); // throttle nav data rate to once per second, until configured
+                        break;
+                    case UBLOX_ACQUIRE_MODEL:
                         ubloxSendNAV5Message(gpsConfig()->gps_ublox_acquire_model);
+                        gpsData.ubloxUsingFlightModel = false;
+                        break;
+                    case UBLOX_CFG_ANA:
+                        if (gpsData.ubloxM7orAbove) { // NavX5 support existed in M5 - in theory we could remove that check
+                            ubloxSendNavX5Message();
+                        } else {
+                            gpsData.state_position++;
+                        }
+                        break;
+                    case UBLOX_MSG_DISABLE_NMEA:
+                        if (gpsData.ubloxM9orAbove) {
```

Comment:
If I counted correctly, this is nest level 4 or 5, and across the levels the function mixed several purposes, blending them into a single, hard/impossible to read or test function.

As above, I strongly suggest splitting up the method into logical blocks.

NB, checks do not belong here (`if (gpsData.ubloxM7orAbove)` or `if (gpsData.ubloxM9orAbove) {` should be moved outside, it is causing excessive clutter here)

An example:
````
case UBLOX_MSG_DISABLE_NMEA:
 disableNMEA();
````

and in `disableNMEA()`
```
if (gpsData.ubloxM9orAbove) {
  ubloxDisableNMEAValSet();
  gpsData.state_position = UBLOX_MSG_RMC; // skip UBX NMEA entries - goes to RMC plus one, or SBAS
  } else {
  gpsData.state_position++;
}
```

Reply:
Agreed.  Once this PR is merged, refactoring PR's will be welcomed.
The focus now is on functional fixes, primarily.
A full refactor is needed but is outside the scope of this PR.